### PR TITLE
[DEV-2736] Implement feast feature store service

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR
* implements feast registry service to create, retrieve & update feast registry;
* implements feast feature store service to construct feast feature store object.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: tests/unit/feast/service/test_registry.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+async def feast_registry_fixture(feast_registry_service, feature_list):
+    """Feast registry fixture"""
+    feature_list_model = feature_list.cached_model
+    registry = await feast_registry_service.create_document(
+        data=FeastRegistryCreate(
+            project_name="test_project",
+            feature_lists=[feature_list_model],
+        )
+    )
+    return registry
```
Comment (gee-senbong) [inline tests/unit/feast/service/test_registry.py:32]: This is how we create feast registry (input is expected to be list of deployed feature list models).

File: tests/unit/feast/service/test_registry.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    assert len(registry_proto.data_sources) == 2
+    assert len(registry_proto.entities) == 2
+
+
+@pytest.mark.asyncio
+async def test_get_feast_feature_store(feast_feature_store_service, feast_registry):
+    """Test get feast feature store"""
+    feast_fs = await feast_feature_store_service.get_feast_feature_store(
+        feast_registry_id=feast_registry.id
+    )
```
Comment (gee-senbong) [inline tests/unit/feast/service/test_registry.py:82]: This is how we construct feast feature store using the feature registry id.

File: featurebyte/feast/utils/registry_construction.py [hunk 3]
```diff
@@ -288,13 +291,50 @@ def create_feast_feature_services(
...
+            for entity in primary_entity_ids_to_feast_entity.values():
+                registry.apply_entity(entity=entity, project=project_name)
+            for feature_view in feast_feature_views:
+                registry.apply_feature_view(feature_view=feature_view, project=project_name)
+            for feature_service in feast_feature_services:
+                registry.apply_feature_service(
+                    feature_service=feature_service, project=project_name
+                )
+            registry_proto = registry.proto()
+            return cast(RegistryProto, registry_proto)
```
Comment (gee-senbong) [inline featurebyte/feast/utils/registry_construction.py:328]: Revise this part as feast feature store auto generates project metadata stores in the registry. Without these metadata, we can't list the assets through feast feature store interface.

File: featurebyte/feast/schema/registry.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+from featurebyte.models.feature_list import FeatureListModel
+from featurebyte.schema.common.base import BaseDocumentServiceUpdateSchema
+
+
+class FeastRegistryCreate(FeatureByteBaseModel):
+    """
+    Feast registry create schema
+    """
+
+    project_name: Optional[str]
```
Comment (YS-L) [inline featurebyte/feast/schema/registry.py:16]: Is it important whether to provide the `project_name` parameter?

File: featurebyte/query_graph/node/schema.py [hunk 5]
```diff
@@ -53,6 +54,30 @@ def create_feast_data_source(
         """
         raise NotImplementedError()
 
+    def get_offline_store_config(self, user_name: Optional[str], password: Optional[str]) -> Any:
```
Comment (YS-L) [inline featurebyte/query_graph/node/schema.py:57]: Would it make sense for the function signature to accept a `BaseDatabaseCredential` to support other credential types in the future?

**Review by YS-L — APPROVED** [review]

File: featurebyte/feast/schema/registry.py (hunk 4 above)
Comment (gee-senbong) [reply featurebyte/feast/schema/registry.py:16]: I don't think it is critical. Just to provide an option here in case we want to override the default one.

**Pull request merged**